src/Mod/TechDraw/App/DimensionFormatter.cpp

```diff
@@ -308,23 +308,60 @@ std::string DimensionFormatter::getFormattedDimensionValue(const int partial) co
 }
 
 // format the value using the formatSpec. Also, handle the non-standard format-
-// specifier '%w', which has the following rules: works as %f, but no trailing zeros
-QString DimensionFormatter::formatValueToSpec(const double value, const QString& formatSpecifier) const
+// specifiers '%w', which works as %f but without trailing zeros
+// and '%r', which round the value to the given precision.
+QString DimensionFormatter::formatValueToSpec(const double value, QString formatSpecifier) const
 {
     QString formattedValue;
-    if (formatSpecifier.contains(QRegularExpression(QStringLiteral("%.*[wW]")))) {
-        QString fs = formatSpecifier;
-        fs.replace(QRegularExpression(QStringLiteral("%(.*)w")), QStringLiteral("%\\1f"));
-        fs.replace(QRegularExpression(QStringLiteral("%(.*)W")), QStringLiteral("%\\1F"));
-        formattedValue = QString::asprintf(fs.toStdString().c_str(), value);
-        // First, try to cut trailing zeros, if AFTER decimal dot there are nonzero numbers
-        // Second, try to cut also decimal dot and zeros, if there are just zeros after it
-        formattedValue.replace(QRegularExpression(QStringLiteral("([0-9][0-9]*\\.[0-9]*[1-9])00*$")), QStringLiteral("\\1"));
-        formattedValue.replace(QRegularExpression(QStringLiteral("([0-9][0-9]*)\\.0*$")), QStringLiteral("\\1"));
-    } else {
+
+    auto format = [](QString f, double value){
+        return QString::asprintf(f.toStdString().c_str(), value);
+    };
+
+    QRegularExpression wrRegExp(QStringLiteral("%(?<dec>.*)(?<spec>[wWrR])"));
+    QRegularExpressionMatch wrMatch = wrRegExp.match(formatSpecifier);
+
+    if (! wrMatch.hasMatch()) {
         if (isNumericFormat(formatSpecifier)) {
-            formattedValue = QString::asprintf(formatSpecifier.toStdString().c_str(), value);
+            formattedValue = format(formatSpecifier, value);
         }
+        return formattedValue;
+    }
+
+    QString spec = wrMatch.captured(QStringLiteral("spec")).toLower();
+    QString dec = wrMatch.captured(QStringLiteral("dec"));
+
+    if (spec == QStringLiteral("w")) {
+        formattedValue = format(QStringLiteral("%") + dec + QStringLiteral("f"), value);
+        // First, cut trailing zeros
+        while(formattedValue.back().toLatin1() == '0')
+        {
+            formattedValue.chop(1);
+        }
+        // Second, try to cut also decimal dot
+        if(formattedValue.back().toLatin1() == '.')
+        {
+            formattedValue.chop(1);
+        }
```

Comment:
```suggestion
        while(formattedValue.endsWith("0")) {
            formattedValue.chop(1);
        }
        // Second, try to cut also decimal dot
        if(formattedValue.endsWith("."))  {
            formattedValue.chop(1);
        }
```

Reply:
Well… It doesn't work because `QT_NO_CAST_FROM_ASCII` is set. So it needs at least
`formattedValue.endsWith(QStringLiteral("0"))`